CRM/ApiLog/ApiLogService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+use Civi\API\Events;
+use Civi\Utils\Settings;
+use Civi\Core\Service\AutoService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+$civicrm_base_path = dirname(__DIR__, 2);
+require $civicrm_base_path . '/vendor/autoload.php';
+
+class CRM_ApiLog_ApiLogService extends AutoService implements EventSubscriberInterface {
+
+  public static function getSubscribedEvents(): array {
```

Comment:
We need to figure out a way to keep track of API requests across these events, so that we only have one log entry per API request.

One approach that we discussed was to add some sort of unique request ID in `onApiPrepare` - it could just be the `ApiLog.id` value - to the event/request, and then use it in the other events (`onApiRespond` or `onApiException`) to update the existing `ApiLog` entry e.g. with response/success status instead of inserting a new record.